client/finality-grandpa/src/environment.rs

```diff
@@ -503,19 +503,19 @@ where
 
 		let is_descendent_of = is_descendent_of(&*self.client, None);
 
-		// TODO: add proper async support here
-		let best_header = futures::executor::block_on(
-			self.select_chain
-				.best_chain()
-				.map_err(|e| Error::Blockchain(e.to_string())),
-		)?;
+		let (best_block_hash, best_block_number) = {
+			// TODO: Use SelectChain::best_chain() to get a
```

Comment:
will it need a linked issue?

Reply:
also, is it important that these two pieces of data might be slightly outdated?

Reply:
I created an issue to track this and added a comment explaining the implications of the change here.